docs/comparisons.rst

```diff
@@ -0,0 +1,132 @@
+.. highlight:: python
+
+Comparisons
+===========
+
+How does marbles compare to other Python testing tools?
+
+For now, we focus on `pytest`_, which has the most in common with
+marbles.
+
+.. toctree::
+   :maxdepth: 2
+
+
+pytest
+------
+
+Marbles provides some functionality similar to `pytest`_:
+
+* Full statement source code presentation
+* Presentation of variables present in assertion statements
+
+Marbles goes beyond pytest with:
+
+* Note annotations
+* Custom assertion mixins
+
+Finally, marbles tests have the same structure as unittest tests,
+which are more explicit than pytest tests, and therefore are easier to
+learn and to reason about at failure time.
+
+If you're familiar with pytest, you'll probably find that writing
+marbles tests is more typing than you're used to, but we hope, no
+matter which tool you know well, responding to a test failure will be
+faster and easier with marbles. In short, pytest is write optimized,
+while marbles is read optimized.
+
+Similarities
+~~~~~~~~~~~~
+
+Assertion Source
+""""""""""""""""
+
+Both marbles and pytest present the :ref:`source code <source>` of the
+whole assertion statement that failed, which is more useful than a
+typical Python traceback.
+
+We believe both tools provide an equivalent benefit here.
+
+Local Variables
+"""""""""""""""
+
+Both marbles and pytest present some of the local variables present at
+the time an assertion in your test failed.
+
+While pytest presents just the variables involved in the assertion
+(and shows each sub-expression involved in the assertion), marbles
+presents all public local variables in scope in the test method.
+
+Marbles is a little more flexible here. While pytest has the ability
+to expand complex expressions and present all of their constituent
+names in the failure message, it doesn't present any other variables
+leading up to the assertion, which may provide valuable context.
+
+With marbles, you can get the same effect as pytest by simply
+decomposing a complex assertion statement into its constituent parts,
+and giving each of these parts a distinct local variable binding.
+
+Consider the following pytest code::
+
+    assert a * b < c * d
+
+If this assertion fails, pytest will show you the values of the
+expressions ``a * b`` and ``c * d``, as well as the individual values
+of each variable ``a``, ``b``, ``c``, and ``d``.
+
+In marbles, one could instead write this test as::
+
+    lhs = a * b
+    rhs = c * d
+    self.assertLess(lhs, rhs)
+
+Advantages of marbles
+~~~~~~~~~~~~~~~~~~~~~
+
+Note annotations
+""""""""""""""""
+
+Marbles gives the test author a place to :ref:`annotate assertion
```

Comment:
This sentence was hard for me to parse. What about: "Marbles allows test authors to :ref:`annotate assertion statements <notes>` with additional information about the test that will help the test consumer put the failed assertion in context."